### ISSUE TITLE
#1: NpmPublishTask

### ISSUE THREAD

**Issue by divramod** [divramod/yagpt#1]
# SUMMARY
As a developer i want to be able to iterate fast and have changes of my npm module available in software in which i use it. So i need to have shortcuts which make it possible for me to automatically do that.

# COMMIT

# TODO
- [x] add semver increment type to args
- [x] create interface for run function
- [ ] create interface for isRunnable function
- [ ] test if npm user is logged in
- [ ] test if test npm package is existant
- [ ] test if user has rights to npm package
- [ ] add creation of changelog (or should that happen manually?)
- [ ] UNpm npmVersionIncrement
- [ ] UNpm npmPublish
- [ ] UNpm isNpmPackage
- [ ] UGit gitPushOriginHEAD

# LATER MILESTONE (move to new issue)
- [ ] unpm: remove prepareNpmRepository and use gitCloneOrCopy instead
- [ ] add function UTask getPathToCurrentProject() --> prevent test to run automatically for current project, because it causes the stash
- [ ] unpm: add function getCurrentPackageName() (for logging issues)
- [ ] unlink package before relinking it (#4: job-npm-publish)
    - [ ] https://stackoverflow.com/questions/19094630/how-do-i-uninstall-a-package-installed-using-npm-
- [ ] document how to make the tests running for other npm users too
- [ ] conditionally skip npm tests if the tests need a test-repository and no npm user is logged in
- [ ] is it possible, that i create a anon user, which can be used in all the npm tests, so that the user logges in before starting the tests and the the user is able to publish to the package?

# Milestone 0.1.0
- [x] rename class?
- [x] document run function
- [x] document isRunnable function
## init cli running (#33)
- [x] test cli commands
- [x] UCommander add
- [x] UCommander.runProgram() add
- [x] mv cli things from index to cli
